x/concentrated-liquidity/incentives_test.go

```diff
@@ -3535,6 +3528,69 @@ func (s *KeeperTestSuite) TestQueryAndClaimAllIncentives() {
 	}
 }
 
+func (s *KeeperTestSuite) TestClaimAllIncentivesForPosition() {
+	// Init suite for the test.
+	s.SetupTest()
+
+	var (
+		defaultAddress   = s.TestAccs[0]
+		defaultBlockTime = time.Unix(1, 1).UTC()
+	)
+
+	s.Ctx = s.Ctx.WithBlockTime(defaultBlockTime)
+	requiredBalances := sdk.NewCoins(sdk.NewCoin(ETH, DefaultAmt0), sdk.NewCoin(USDC, DefaultAmt1))
+	s.FundAcc(defaultAddress, requiredBalances)
+
+	// Create CL pool
+	pool := s.PrepareConcentratedPool()
+
+	// Set up position
+	positionIdOne, _, _, _, _, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, pool.GetId(), defaultAddress, DefaultCoins, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
+	s.Require().NoError(err)
+
+	// Set incentives for pool to ensure accumulators work correctly
+	testIncentiveRecord := types.IncentiveRecord{
+		PoolId:               pool.GetId(),
+		IncentiveDenom:       USDC,
+		IncentiveCreatorAddr: s.TestAccs[0].String(),
+		IncentiveRecordBody: types.IncentiveRecordBody{
+			RemainingAmount: sdk.NewDec(100000),
+			EmissionRate:    sdk.NewDec(1), // 1 per second
+			StartTime:       s.Ctx.BlockTime(),
+		},
+		MinUptime: time.Nanosecond,
+	}
+	err = s.App.ConcentratedLiquidityKeeper.SetMultipleIncentiveRecords(s.Ctx, []types.IncentiveRecord{testIncentiveRecord})
+	s.Require().NoError(err)
+
+	// attempting to claim with same blocktime
+	collectedInc, _, err := s.App.ConcentratedLiquidityKeeper.ClaimAllIncentivesForPosition(s.Ctx, positionIdOne)
+	s.Require().NoError(err)
+
+	// expected collectedIncentives in same blockTime = 0
+	s.Require().Equal(sdk.NewCoins(), collectedInc)
+
+	// add 1 hour to current blocktime
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(time.Minute))
+
+	// attempting to claim after 1min has passed
+	collectedInc, _, err = s.App.ConcentratedLiquidityKeeper.ClaimAllIncentivesForPosition(s.Ctx, positionIdOne)
+	s.Require().NoError(err)
+
+	// expected collectedIncentives after 1 hour  = 59.999999999901820104usdc ~ 59usdc
+	s.Require().Equal(sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(59))), collectedInc)
+
+	// add 1 hour to current blocktime
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(time.Hour))
+
+	// attempting to claim after 1hr has passed
+	collectedInc, _, err = s.App.ConcentratedLiquidityKeeper.ClaimAllIncentivesForPosition(s.Ctx, positionIdOne)
+	s.Require().NoError(err)
+
+	// expected collectedIncentives after 1 hour = 3599.999999998662853243usdc ~ 3599usdc
+	s.Require().Equal(sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(3599))), collectedInc)
+}
+
```

Comment:
Can we also check the balance for the community pool?

Reply:
do we send funds to the community pool here? i thought we just calculate incentives based on unclaimedTotal in accumulator

Reply:
aah i see, this was for forfeited incentives, which we technically donot handle because of 1ns uptime